packages/service-worker/safety-worker.js

```diff
@@ -18,3 +18,16 @@ self.addEventListener('activate', event => {
     console.log('NGSW Safety Worker - unregistered old service worker');
   });
 });
+
+caches.keys().then(cacheKeys =>
+  Promise.all(
+    cacheKeys.map(cacheKey => {
+      const ngswRegex = /^(ngsw).*/;
+      if (ngswRegex.test(cacheKey)) {
+        return caches
+          .open(cacheKey)
+          .then(cache => cache.keys().then(requests => requests.map(req => cache.delete(req))));
+      }
+    })
+  )
+);
```

Comment:
A few suggestions/improvements:

- It is safer to delete the caches inside the `activate` event handler (to avoid deleting the caches before this SW takes over).
- There is no need to delete each request in a cache. We can just delete the whole cache.
- We should be targeting caches that start with `ngsw:` (notice the double-colon at the end).
- We should ensure the SW instance remains active while we delete the caches (by passing a promise to `event.waitUntil()`.

Based on the above, the code look like this (unless I've missed something 😅):

```suggestion
event.waitUntil(caches.keys().then(cacheNames => {
  const ngswCacheNames = cacheNames.filter(name => /^ngsw:/.test(name));
  return Promise.all(ngswCacheNames.map(name => caches.delete(name)));
}));
```